src/inmanta/plugins.py

```diff
@@ -294,6 +294,12 @@ def to_dsl_type(python_type: type[object]) -> inmanta_type.Type:
             bases = [to_dsl_type(arg) for arg in typing.get_args(python_type)]
             return inmanta_type.Union(bases)
 
+    if dataclasses.is_dataclass(python_type):
+        entity = get_inmanta_type_for_dataclass(python_type)
+        if entity:
+            return entity
+        raise TypingException(None, f"invalid type {python_type}, this dataclass has no associated inmanta entity")
```

Comment:
I'm a bit uncertain about this, I love to close the gap, but I'm afraid of what will blow up

Reply:
But native annotations are a new feature, nothing can blow up really, can it?